tests/posix/pthread_equal/src/pthread_equal.c

```diff
@@ -22,12 +22,19 @@ void *thread_top(void *p1)
 
 void test_pthread_equal(void)
 {
-	int ret = 1;
+	int ret = -1;
 	pthread_attr_t attr;
 	struct sched_param schedparam;
 	pthread_t newthread;
 
-	pthread_attr_init(&attr);
+	ret = pthread_attr_init(&attr);
+	if (ret != 0) {
+		zassert_false(pthread_attr_destroy(&attr),
```

Comment:
Is there any point in calling pthread_attr_destroy() if ret is not 0? pthread_attr_destroy() can be called regardless of success from pthread_attr_init(); otherwise, it's going to be called with a potentially invalid pthread_attr_t struct (won't crash but won't signal an error either).

Reply:
Hi @lpereira , 

pthread_attr_init is going to fail only on one condition that - attrib object is already initialized. If attrib object is initialized then it need to be destroyed. Because of that only we are calling pthread_attr_destroy and initializing the attrib object again.
as attrib->status is just a variable and it may have garbage value, so this garbage value may indicate that this attr object is initialized. So to handle those cases I have put this check.